### PR TITLE
Support custom rsync flags.

### PR DESCRIPTION
**Pull request by mattstein**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

The now-deprecated [rsync recipe](https://github.com/deployphp/recipes/blob/master/recipe/rsync.php) only passed the `-rz` flags and offered the option to override them. This PR restores the ability to pass custom flags, making it possible to update legacy `rsync` tasks to use `upload` without being broken by the currently-hardcoded `-azP` flags.

**Legacy example using `rsync` task:**

```php
require 'recipe/rsync.php';

$exclude = [
    '.git',
    '.babelrc',
    '/.gitignore',
    '/.env',
    '/.env.example',
    '/node_modules/',
    '.github',
    'deploy.php'
];

add('rsync', [ 'exclude' => $exclude ]);
set('rsync_src', function () {
    return __DIR__;
});
```

**Identical functionality with `upload` task and PR for `flags` option:**

```php
$exclude = [
    '.git',
    '.babelrc',
    '/.gitignore',
    '/.env',
    '/.env.example',
    '/node_modules/',
    '.github',
    'deploy.php'
];

set('upload_exclude', $exclude);

task('upload', function () {
    $options = [
        '--delete'
    ];

    foreach (get('upload_exclude') as $item) {
        $options[] = sprintf("--exclude='%s'", $item);
    }

    upload(
        __DIR__ . '/',
        '{{release_path}}/',
        ['options' => $options, 'flags' => 'rz']
    );
});
```

### CODE REVIEW

**Comment by antonmedv** [comment]
I'm closing this PR for now in order to housekeeping. Fill free to reopen if needed.

File: src/Utility/Rsync.php [hunk 1]
```diff
@@ -34,10 +34,12 @@ public function call(Host $host, string $source, string $destination, array $con
         $defaults = [
             'timeout' => null,
             'options' => [],
+            'flags'   => 'azP'
         ];
         $config = array_merge($defaults, $config);
 
         $options = $config['options'] ?? [];
```
[the remark this answers is not in the document]
Comment (mattstein) [reply src/Utility/Rsync.php:41]: @antonmedv You can use options to pass _additional_ flags, but that `-azP` is currently hardcoded and cannot be changed by a subsequent flag. That’s what this changes.

Perhaps I’m misunderstanding you—if this PR is unnecessary and I want to call `rsync` with only `-rz`, how would I do that using the current `upload()` task?

[the remark this answers is not in the document]
Comment (antonmedv) [reply src/Utility/Rsync.php:41]: Oh! Got it! Thanks for explanation.

**Comment by antonmedv** [comment]
@mattstein can you also add doc about this functionality? Just in comments to task/options. And run bin/docgen command.

**Comment by mattstein** [comment]
Will do @antonmedv. 👍

**Pull request merged**